nios lookup errors out when there are no results #37970

justjais opened the pull request:
##### SUMMARY
When using the nios lookup feature. If a lookup has no results it fails out. with "'NoneType' object is not iterable"

##### ISSUE TYPE
Bug Report

##### COMPONENT NAME
nios_lookup

##### ANSIBLE VERSION
ansible 2.6.0 (devel 2d0547c411)
python version = 2.7.14 (default, Mar 10 2018, 00:01:04) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]


##### ADDITIONAL INFORMATION
OS / ENVIRONMENT
MAC 10.12.6

```

justjais committed:
Update nios.py

justjais committed:
Update nios.py

justjais committed:
Update nios.py

justjais committed:
nios lookup errors out when there are no results #37970  Open	

Indentation failure issue resolved

ganeshrn commented on lib/ansible/plugins/lookup/nios.py:
```diff
@@ -108,7 +108,8 @@ def run(self, terms, variables=None, **kwargs):
+            for obj in res:
+                if 'extattrs' in obj:
+                    obj['extattrs'] = flatten_extattrs(obj['extattrs'])
         return res
```
If the return object is None should the lookup plugin return an empty list instead of None?

justjais replied:
Yes, we should return empty list instead of None. Also, wanted to confirm that should I initialise the res with empty list or in else block return an empty list?

justjais replied:
Yes, empty list should be returned instead of None, will update the changes and checkin again

ganeshrn reviewed (commented):
Also `return` doc need to be updated to cover the senecario for no result.

justjais committed:
Returning empty list instead of None

In case of no results, res will be returned as an empty list instead of None (implementing ganeshrn comment)

gundalow commented:
There was an issue that got introduced into devel during the test update. I've triggered a rerun of CI which should pass.
